### PR TITLE
Feature: Add OpenAPI3 doc for user_capacities endpoint.

### PR DESCRIPTION
**Pull request by aymebou**
Adding documentation for new endpoint we are currently developing.

A few remarks on POST and PUT requests : 

- We might want to specify that some params (eg: L5204-L5221) can be used several times.

- We might want to explain about the orders of the params :

A request with params
name:Cap
channels[][channel_id]:id_1
channels[][default_capacity]:1
channels[][max_capacity]:2
channels[][channel_id]:id_2
channels[][default_capacity]:1
channels[][max_capacity]:10

And with params (just switching orders) :

name:Cap
channels[][channel_id]:id_1
channels[][channel_id]:id_2
channels[][default_capacity]:1
channels[][max_capacity]:2
channels[][default_capacity]:1
channels[][max_capacity]:10

Will NOT yield the same results (actually the second one will be rejected).

I tried to mention it on the documentation but it might be unclear, so I wanted to bring that up.

[Ticket](https://jira.ringcentral.com/browse/RD-15138)

### CODE REVIEW

File: specs/engage-digital_openapi3.yaml
Comment (DrissTM) [inline specs/engage-digital_openapi3.yaml:5204]: Not sure about this one, there's probably a better way to say this
```suggestion
          - description: The id of the channel on which the capacities applies.
```

I'll let you have a look at the other occurrences

File: specs/engage-digital_openapi3.yaml
Comment (DrissTM) [inline specs/engage-digital_openapi3.yaml:5232]: IMO we should display the routes in this order:

- GET /user_capacities
- GET /user_capacities/:id
- POST /user_capacities
- PUT /user_capacities/:id
- DELETE /user_capacities/:id

File: specs/engage-digital_openapi3.yaml
Comment (DrissTM) [inline specs/engage-digital_openapi3.yaml:5193]: If I understood correctly this one refers to the id that we need to put in the url in some routes (GET, PUT, DELETE) but this is not valid for POST

**Review by DrissTM — CHANGES_REQUESTED** [review]

File: specs/engage-digital_openapi3.yaml
Comment (aymebou) [reply specs/engage-digital_openapi3.yaml:5232]: Actually I don't think it is possible since they need to be grouped by url.

Comment (DrissTM) [reply specs/engage-digital_openapi3.yaml:5232]: oh okay then :+1:

**Comment by nel** [comment]
@aymebou is there a way to have a preview of this Branch/MR doc ? AFAICT the js I did on https://ringcentral.github.io/engage-digital-api-docs/digital/ only support master.

Would there be a nice way to visualize result ? And you 2 guys have you actually visualized results or just did yaml editing ?

**Comment by aymebou** [comment]
I found https://editor.swagger.io/ which we can use to view the yaml before merging actually.

File: specs/engage-digital_openapi3.yaml
Comment (aymebou) [reply specs/engage-digital_openapi3.yaml:5227]: And actually on other endpoints the order seems to be on alphabetical order : 

```
- GET      /user_capacities
- POST     /user_capacities
- DELETE   /user_capacities/:id
- GET      /user_capacities/:id
- PUT      /user_capacities/:id
```

I switched it to match this.

File: specs/engage-digital_openapi3.yaml
Comment (DrissTM) [inline specs/engage-digital_openapi3.yaml:5306]:
```suggestion
        - description: The maximum capacity to apply to the related channel.
```

File: specs/engage-digital_openapi3.yaml
Comment (DrissTM) [inline specs/engage-digital_openapi3.yaml:5205]:
```suggestion
          - description: The default capacity to apply to the related channel.
```

File: specs/engage-digital_openapi3.yaml
Comment (DrissTM) [inline specs/engage-digital_openapi3.yaml:80]: only one newline is probably enough
```suggestion
```

File: specs/engage-digital_openapi3.yaml
Comment (DrissTM) [inline specs/engage-digital_openapi3.yaml:5253]:
```suggestion
        This method renders the user capacity corresponding to the id given in parameter.
```

We should also mention that it renders a 404 if the record can't be found

**Review by DrissTM — CHANGES_REQUESTED** [review]

File: specs/engage-digital_openapi3.yaml
Comment (aymebou) [reply specs/engage-digital_openapi3.yaml:27]: yes it should indeed

File: specs/engage-digital_openapi3.yaml
Comment (DrissTM) [inline specs/engage-digital_openapi3.yaml:7786]: `id`, `created_at` and `updated_at` are missing here

**Review by DrissTM — CHANGES_REQUESTED** [review]

**Comment by DrissTM** [comment]
@CedricBm I think you can have a look now, besides my last comment I think we're good

File: specs/engage-digital_openapi3.yaml
Comment (CedricBm) [inline specs/engage-digital_openapi3.yaml:5281]: Same remark for the newline + code formatting

File: specs/engage-digital_openapi3.yaml
Comment (CedricBm) [inline specs/engage-digital_openapi3.yaml:6959]: Shouldn't this one be called `GetAllUserCapacitiesResponse`?

**Review by CedricBm — CHANGES_REQUESTED** [review]

File: specs/engage-digital_openapi3.yaml
Comment (aymebou) [reply specs/engage-digital_openapi3.yaml:7786]: Nice catch, and the props should be in alphabetical order

File: specs/engage-digital_openapi3.yaml
Comment (DrissTM) [reply specs/engage-digital_openapi3.yaml:5200]: actually I made a mistake on this one, capacities shouldn't be plural:
```suggestion
          - description: The id of the channel on which the capacity applies.
```

Comment (aymebou) [reply specs/engage-digital_openapi3.yaml:5200]: Oh no it should, since you can have several capacities inside this array

Comment (DrissTM) [reply specs/engage-digital_openapi3.yaml:5200]: ah yes if you consider the default and max it should be plural, I was thinking about the object in ED. let's apply Cedric's suggestion then

File: specs/engage-digital_openapi3.yaml
Comment (CedricBm) [inline specs/engage-digital_openapi3.yaml:5291]: Any reason to have a 4 space indentation here?

File: specs/engage-digital_openapi3.yaml
Comment (CedricBm) [inline specs/engage-digital_openapi3.yaml:5186]: Any reason to have a 4 space indentation here?

**Review by CedricBm — CHANGES_REQUESTED** [review]

File: specs/engage-digital_openapi3.yaml
Comment (DrissTM) [inline specs/engage-digital_openapi3.yaml:5180]:
```suggestion
                $ref: '#/components/schemas/GetAllUserCapacitiesResponse'
```

File: specs/engage-digital_openapi3.yaml
Comment (DrissTM) [inline specs/engage-digital_openapi3.yaml:5252]:
```suggestion
                $ref: '#/components/schemas/UserCapacity'
```

**Review by DrissTM — CHANGES_REQUESTED** [review]

**Review by CedricBm — APPROVED** [review]
LGTM

**Review by DrissTM — APPROVED** [review]
LGTM

**Pull request merged**
